src/instamatic/camera/videostream.py

```diff
@@ -109,43 +130,48 @@ def __getattr__(self, attrname):
     def start(self):
         self.grabber.start_loop()
 
-    def send_frame(self, frame, acquire=False):
-        if acquire:
-            self.grabber.lock.acquire(True)
-            self.acquired_frame = self.frame = frame
-            self.grabber.lock.release()
-            self.grabber.acquireCompleteEvent.set()
-        else:
-            self.grabber.lock.acquire(True)
-            self.frame = frame
-            self.grabber.lock.release()
-
-    def setup_grabber(self) -> ImageGrabber:
-        grabber = ImageGrabber(self.cam, callback=self.send_frame, frametime=self.frametime)
+    def send_media(
+        self,
+        media: Union[np.ndarray, List[np.ndarray]],
+        request: Optional[MediaRequest] = None,
+    ) -> None:
+        """Callback function of `self.grabber` that handles grabbed media."""
+        with self.grabber.lock:
+            if request is None:
+                self.frame = media
+            elif isinstance(request, ImageRequest):
+                self.requested_media = self.frame = media
+                self.grabber.acquireCompleteEvent.set()
+            else:  # isinstance(request, MovieRequest):
+                self.requested_media = media
+                self.frame = media[-1]
+                self.grabber.acquireCompleteEvent.set()
```

Comment:
Shuffling these around to make the default (None) be more logical.

```suggestion
            if isinstance(request, ImageRequest):
                self.requested_media = self.frame = media
                self.grabber.acquireCompleteEvent.set()
            elif isinstance(request, MovieRequest):
                self.requested_media = media
                self.frame = media[-1]
                self.grabber.acquireCompleteEvent.set()
            else:
                self.frame = media
```

Reply:
They are ordered according to their probability; this ordering is the most logical to me because typically (i.e. 20 times per second) just one check is necessary rather than three.